### PR TITLE
Feature/update spring boot bom

### PR DESCRIPTION
**Pull request by zhoufenqin**
release com.microsoft.azure:spring-starter-azure-storage bom

### CODE REVIEW

File: sdk/boms/azure-spring-boot-bom/pom.xml [hunk 1]
```diff
@@ -95,7 +94,12 @@
       <dependency>
         <groupId>com.microsoft.azure</groupId>
         <artifactId>azure-servicebus-jms-spring-boot-starter</artifactId>
-        <version>${azure.spring.boot.jms.integration.version}</version>
+        <version>${azure.spring.boot.integration.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>com.microsoft.azure</groupId>
+        <artifactId>spring-starter-azure-storage</artifactId>
+        <version>${azure.spring.boot.integration.version}</version>
```
Comment (saragluna) [inline sdk/boms/azure-spring-boot-bom/pom.xml:102]: This version should be different from the one of starters?

**Review by saragluna — CHANGES_REQUESTED** [review]
Overall is great, but the version for storage starter is supposed to be different from others.

File: sdk/boms/azure-spring-boot-bom/pom.xml (hunk 1 above)
Comment (zhoufenqin) [reply sdk/boms/azure-spring-boot-bom/pom.xml]: done. updated

File: sdk/boms/azure-spring-boot-bom/pom.xml [hunk 2]
```diff
@@ -95,7 +95,12 @@
       <dependency>
         <groupId>com.microsoft.azure</groupId>
         <artifactId>azure-servicebus-jms-spring-boot-starter</artifactId>
-        <version>${azure.spring.boot.jms.integration.version}</version>
+        <version>${azure.spring.boot.integration.version}</version>
```
Comment (chenrujun) [inline sdk/boms/azure-spring-boot-bom/pom.xml:98]: The original version of `azure-servicebus-jms-spring-boot-starter` (2.3.4) is different from `project version` (2.3.3), is it OK to set to `project version` directly?

File: sdk/spring/azure-spring-boot-starter/CHANGELOG.md [hunk 3]
```diff
@@ -4,27 +4,26 @@
 
 
 ## 2.3.5 (2020-09-14)
-_Improved_
+### Breaking Changes
 - Unify spring-boot-starter version
 
 ## 2.3.4 (2020-08-20)
-_Key Bug Fixes_
+###Key Bug Fixes
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter/CHANGELOG.md:11]: `### Key Bug Fixes`

File: sdk/boms/azure-spring-boot-bom/pom.xml [hunk 4]
```diff
@@ -38,7 +38,7 @@
     <packageOutputDirectory>${project.build.directory}</packageOutputDirectory>
     <legal><![CDATA[[INFO] Any downloads listed may be third party software.  Microsoft grants you no rights for third party software.]]></legal>
     <azure.spring.boot.integration.version>${project.version}</azure.spring.boot.integration.version>
-    <azure.spring.boot.jms.integration.version>2.3.4</azure.spring.boot.jms.integration.version>
+    <azure.spring.boot.storage.integration.version>1.2.8</azure.spring.boot.storage.integration.version>
```
Comment (chenrujun) [inline sdk/boms/azure-spring-boot-bom/pom.xml:41]: `azure.spring.boot.jms.integration.version` is still used in this file. For example line 62 and line 67

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: sdk/boms/azure-spring-boot-bom/pom.xml (hunk 2 above)
Comment (zhoufenqin) [reply sdk/boms/azure-spring-boot-bom/pom.xml:98]: I have already release spring-boot-starter 2.3.5 version, see https://github.com/Azure/azure-sdk-for-java/pull/15146

File: sdk/boms/azure-spring-boot-bom/pom.xml (hunk 4 above)
Comment (zhoufenqin) [reply sdk/boms/azure-spring-boot-bom/pom.xml:41]: done

File: sdk/spring/azure-spring-boot-starter/CHANGELOG.md (hunk 3 above)
Comment (zhoufenqin) [reply sdk/spring/azure-spring-boot-starter/CHANGELOG.md]: done

**Review by saragluna — APPROVED** [review]
LGTM.

**Review by chenrujun — APPROVED** [review]
LGTM.

**Pull request merged**